packages/lib/notion/service.ts

```diff
@@ -0,0 +1,105 @@
+import { prisma } from "@formbricks/database";
+import { TNotionConfig, TNotionDatabase, TNotionIntegration } from "@formbricks/types/v1/integrations";
+import { cache } from "react";
+import { Prisma } from "@prisma/client";
+import { DatabaseError } from "@formbricks/types/v1/errors";
+import { symmetricDecrypt } from "../crypto";
+import { ENCRYPTION_KEY } from "../constants";
+
+async function fetchPages(config: TNotionIntegration["config"]) {
+  try {
+    const res = await fetch("https://api.notion.com/v1/search", {
+      headers: getHeaders(config),
+      method: "POST",
+      body: JSON.stringify({
+        page_size: 100,
+        filter: {
+          value: "database",
+          property: "object",
+        },
+      }),
+    });
+    return (await res.json()).results;
+  } catch (err) {
+    throw err;
+  }
+}
+
+export const getNotionDatabases = async (environmentId: string): Promise<TNotionDatabase[]> => {
+  let results: TNotionDatabase[] = [];
+  try {
+    const notionIntegration = await getNotionIntegration(environmentId);
+    if (notionIntegration && notionIntegration.config?.key.bot_id) {
+      results = await fetchPages(notionIntegration.config);
+    }
+    return results;
+  } catch (error) {
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      throw new DatabaseError("Database operation failed");
+    }
+    throw error;
+  }
+};
+
+export const getNotionIntegration = cache(
```

Comment:
please check the latest changes from main. We now use the integration service to get the integration by type and only have the service for the integration, e.g. here the notion service, to communicate with notion